pypeit/core/coadd2d.py

```diff
@@ -854,11 +855,21 @@ def load_coadd2d_stacks(self, spec2d_files):
         tslits_dict_list = []
         # TODO Sort this out with the correct detector extensions etc.
         # Read in the image stacks
+        waveimgfile, tiltfile, tracefile = None, None, None
         for ifile in range(nfiles):
-            #waveimg = WaveImage.load_from_file(waveimgfiles[ifile])  # JXP
-            waveimg = WaveImage.from_master_file(waveimgfiles[ifile]).image
-            #tilts = WaveTilts.load_from_file(tiltfiles[ifile])
-            tilts = WaveTilts.from_master_file(tiltfiles[ifile]).tilts_dict
+            # Load up the calibs, if needed
+            if waveimgfiles[ifile] == waveimgfile:
+                pass
+            else:
+                waveimg = WaveImage.from_master_file(waveimgfiles[ifile]).image
```

Comment:
agreed.  I think I had a reason for this,
but I have long forgotten what it was.

Have edited it to your suggested approach.

Reply:
I still see the pass in there, so it does not look like Kyle's suggestion was implemented. Is the point of this to avoid reading the same calibraitons over and over again if they indeed point to the same file?

Reply:
I still see the pass in there, so it does not look like Kyle's suggestion was implemented. Is the point of this to avoid reading the same calibraitons over and over again if they indeed point to the same file?

Reply:
I still see the pass in there, so it does not look like Kyle's suggestion was implemented. Is the point of this to avoid reading the same calibraitons over and over again if they indeed point to the same file?

Reply:
I still see the pass in there, so it does not look like Kyle's suggestion was implemented. Is the point of this to avoid reading the same calibraitons over and over again if they indeed point to the same file?